internal/provider/resource_ably_namespace.go

```diff
@@ -0,0 +1,278 @@
+func (r resourceNamespace) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	// Checks whether the provider and API Client are configured. If they are not, the provider responds with an error.
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider hasn't been configured before apply",
+		)
+		return
+	}
+
+	// Gets plan values
+	var plan AblyNamespace
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Generates an API request body from the plan values
+	namespace_values := ably_control_go.Namespace{
+		ID:               plan.ID.Value,
+		Authenticated:    plan.Authenticated.Value,
+		Persisted:        plan.Persisted.Value,
+		PersistLast:      plan.PersistLast.Value,
+		PushEnabled:      plan.PushEnabled.Value,
+		TlsOnly:          plan.TlsOnly.Value,
+		ExposeTimeserial: plan.ExposeTimeserial.Value,
+	}
+
+	// Creates a new Ably namespace by invoking the CreateNamespace function from the Client Library
+	ably_namespace, err := r.p.client.CreateNamespace(plan.AppID.Value, &namespace_values)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating Resource",
+			"Could not create resource, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	// Maps response body to resource schema attributes.
+	resp_apps := AblyNamespace{
+		AppID:            types.String{Value: plan.AppID.Value},
+		ID:               types.String{Value: ably_namespace.ID},
+		Authenticated:    types.Bool{Value: ably_namespace.Authenticated},
+		Persisted:        types.Bool{Value: ably_namespace.Persisted},
+		PersistLast:      types.Bool{Value: ably_namespace.PersistLast},
+		PushEnabled:      types.Bool{Value: ably_namespace.PushEnabled},
+		TlsOnly:          types.Bool{Value: ably_namespace.TlsOnly},
+		ExposeTimeserial: types.Bool{Value: namespace_values.ExposeTimeserial},
+	}
+
+	// Sets state for the new Ably App.
+	diags = resp.State.Set(ctx, resp_apps)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read resource
+func (r resourceNamespace) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+	// Gets the current state. If it is unable to, the provider responds with an error.
+	var state AblyNamespace
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Gets the Ably App ID and namespace ID value for the resource
+	app_id := state.AppID.Value
+	namespace_id := state.ID.Value
+
+	// Fetches all Ably Namespaces in the app. The function invokes the Client Library Namespaces() method.
+	// NOTE: Control API & Client Lib do not currently support fetching single namespace given namespace id
+	namespaces, err := r.p.client.Namespaces(app_id)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating Resource",
```

Comment:
I feel like these should be way more specific strings. But copied the ones over from app for now.

Reply:
Just saw this now - its exactly what I commented on in my review :)